bindings/go/oci/repository/provider/credential_cache.go

```diff
@@ -76,18 +54,11 @@ func (cache *credentialCache) add(spec *ocirepospecv1.Repository, credentials ma
 	cache.mu.Lock()
 	defer cache.mu.Unlock()
 
-	parsedBaseURL, err := url.Parse(spec.BaseUrl)
+	identity, err := runtime.ParseURLToIdentity(spec.BaseUrl)
```

Comment:
in the ghrc.io example, we would have "dummy" instead of "https" in the scheme which would be cleared afterwards - is that desired?

I guess that `ghrc.io`  should be cached under the `https` scheme in case we have an entry with https and one without.

```go
func ParseURLAndAllowNoScheme(urlToParse string) (*url.URL, error) {
	const dummyScheme = "dummy"
	if !strings.Contains(urlToParse, "://") {
		urlToParse = dummyScheme + "://" + urlToParse
	}
	parsedURL, err := url.Parse(urlToParse)
	if err != nil {
		return nil, err
	}
	if parsedURL.Scheme == dummyScheme {
		parsedURL.Scheme = ""
	}
	return parsedURL, nil
}
```

Reply:
ghrc.io is just an example - my assumption is that if not scheme or port is provided, we would fallback to https - that's at least what feels most natural for me from a view of a user

Reply:
in the example `ghcr.io` we would land with an identity without scheme because `dummy` is removed in the method. The "fallback" you describe is the default for the OCI distribution spec. It is then the clients responsibility to use the correct scheme (ORAS does this already)

Reply:
gotcha . if this it the goal, I guess the change here would work just fine